pkgs/applications/audio/sonic-pi/default.nix

```diff
@@ -71,17 +67,16 @@ mkDerivation rec {
     popd
 
     pushd app/gui/qt
-      cp -f ruby_help.tmpl ruby_help.h
-      ../../server/ruby/bin/qt-doc.rb -o ruby_help.h
+      cp -f utils/ruby_help.tmpl utils/ruby_help.h
+      ../../server/ruby/bin/qt-doc.rb -o utils/ruby_help.h
 
-      substituteInPlace SonicPi.pro \
-        --replace "LIBS += -lrt -lqt5scintilla2" \
-                  "LIBS += -lrt -lqscintilla2 -lqwt"
+      lrelease lang/*.ts
 
-      lrelease SonicPi.pro
-      qmake SonicPi.pro
-
-      make
+      mkdir build
+      pushd build
+        cmake -G "Unix Makefiles" ..
```

Comment:
I know that is not changed in this PR, but shouldn't the call to CMake go in configurePhase? If you know a technical reason why it can not be in configurePhase, maybe it deserves a comment.

Reply:
Thanks for your review!

I tried moving the CMake to the `configurePhase`, but it needs the `/build/source/app/gui/qt/help_files.qrc` that is build using the `server/ruby/bin/qt-doc.rb` script, which requires `ruby`.. and apparently `buildInputs` are not available in the `configurePhase`.

Reply:
Wait, what? That breaks my understanding of Nix.

Reply:
Scratch my comment about `ruby` not being present in the configurePhase, it is. I misread the logs. What is required is a file `2.6/ffi_c` which appears to be compiled by `./compile-extensions.rb`. It also needs the environment variables `SONIC_PI_HOME` and `AUBIO_LIB` ..

I agree that `CMake` should be in the `configurePhase`, but hauling in all the other stuff from the `buildPhase` feels wrong. What do you think?

Reply:
Seems fine with me in the buildPhase